src/fields/field_fe.cc

```diff
@@ -239,8 +239,21 @@ void FieldFE<spacedim, Value>::interpolate(ElementDataCache<double>::ComponentDa
 		std::fill(elem_count.begin(), elem_count.end(), 0);
 		for (std::vector<unsigned int>::iterator it = searched_elements.begin(); it!=searched_elements.end(); it++) {
 			ElementFullIter elm = source_mesh->element( *it );
-			arma::mat map = elm->element_map();
-			arma::vec projection = elm->project_point(ele->centre(), map);
+			arma::mat map;
+			arma::vec projection;
+			switch (elm->dim()) {
+			case 1:
+				value_handler1_.point_projection(ele->centre(), projection, map, *elm);
+				break;
+			case 2:
+				value_handler2_.point_projection(ele->centre(), projection, map, *elm);
+				break;
+			case 3:
+				value_handler3_.point_projection(ele->centre(), projection, map, *elm);
```

Comment:
Method that decide if the point is in element or not. Move next if into the method and remove second test (projection close to centre). Even better move this decision method into Mapping.